Origin/feature/aws deploy using dns and https

danielmiu opened the pull request:
### Requirements for making a pull request

Thank you for contributing to our project! 

Please fill out the template below to help the project maintainers review it as fast as possible and include your contribution to the project.

### What does it fix?

Closes <!-- If it is related to an open issue, please link the issue here. -->

Please mention the main changes this PR brings.

### How has it been tested?

Please describe the tests that you ran to verify your changes.

<!-- If applicable, mention any known issues with the pull request -->

danielmiu committed:
added DNS and HTTPS

danielmiu committed:
fix conflict

danielmiu committed:
add region var

RaduCStefanescu commented:
Just a check up on my side.
I don't see the `EmailService` and the `JobScheduler` being deployed, although they are integral part of the application, and they both have Dockerfiles.
Let me know @danielmiu

danielmiu committed:
Added emailservice,jobscheduler and deploy.yml

Added EmailService and JobScheduler docker/dns/certificate configuration
Created deploy.yml for GHA deploy
Changed code and variable to accept terraform.workspace as variable
Postgres uses for now image directly from docker

danielmiu commented:
added emailservice and jobscheduler deployment scripts;
created deploy.yml for GHA deployment;
postgres docker uses image from dockerhub directly until more information is shared if dockerfile will be created also for this component.

**Following secrets are required in GITHUB**
DOCKER_HUB_ORGANIZATION
DOCKER_HUB_USERNAME
DOCKER_HUB_PASSWORD
AWS_ACCESS_KEY_ID - _send me a message on slack so I can share it_
AWS_SECRET_ACCESS_KEY - _send me a message on slack so I can share it_
Please create secrets that will be passed to EmailService Docker as variable. After sharing this information, I will create another PR to create the environments.

bvizureanu committed:
Merge branch 'develop' into origin/feature/aws_deploy_using_dns_and_https

RaduCStefanescu committed:
Merge branch 'develop' into origin/feature/aws_deploy_using_dns_and_https

RaduCStefanescu committed:
Merge branch 'develop' into origin/feature/aws_deploy_using_dns_and_https
